### PR TITLE
[paws-collector] Review collector remediations / status for O365

### PR DESCRIPTION
**Pull request by surajkadam**
Review remediaitons and status updates for O365 collectors and add new remediations where necessary

### CODE REVIEW

**Review by OHaganA — APPROVED** [review]

File: collectors/o365/o365_collector.js [hunk 1]
```diff
@@ -174,6 +174,17 @@ class O365Collector extends PawsCollector {
                 try {
                     let error = JSON.parse(message.slice(message.indexOf('{'), message.lastIndexOf('}') + 1));
                     err.errorCode = error.error;
+                    if (error.error_codes) {
```
Comment (mtrybulec) [inline collectors/o365/o365_collector.js:177]: Do we want to return the error code in the message (if customer/support want to search on the error code)?

File: collectors/o365/o365_collector.js [hunk 2]
```diff
@@ -174,6 +174,17 @@ class O365Collector extends PawsCollector {
...
+                        if (error.error_codes[0] === 7000215) {
+                            return callback("Invalid client secret is provided.");
+                        }
+                        if (error.error_codes[0] === 700016) {
+                            return callback("Invalid client ID is provided.");
+                        }
+                        if (error.error_codes[0] === 90002) {
+                            return callback("Please make sure you have the correct tenant ID or This may happen if there are no active subscriptions for the tenant. Check with your subscription administrator.");
```
Comment (mtrybulec) [inline collectors/o365/o365_collector.js:185]: "this" instead of "This"?

File: collectors/o365/o365_collector.js (hunk 1 above)
Comment (surajkadam) [reply collectors/o365/o365_collector.js:177]: okay will do that.

File: collectors/o365/o365_collector.js (hunk 2 above)
Comment (surajkadam) [reply collectors/o365/o365_collector.js:185]: sure

Comment (surajkadam) [reply collectors/o365/o365_collector.js]: Done

File: collectors/o365/o365_collector.js (hunk 1 above)
Comment (surajkadam) [reply collectors/o365/o365_collector.js:177]: Done

**Pull request merged**